server/src/main/java/io/druid/server/ClientInfoResource.java

```diff
@@ -62,19 +63,24 @@
   private static final String KEY_DIMENSIONS = "dimensions";
   private static final String KEY_METRICS = "metrics";
 
-  private static final int SEGMENT_HISTORY_MILLIS = 7 * 24 * 60 * 60 * 1000; // ONE WEEK
-
   private InventoryView serverInventoryView;
   private TimelineServerView timelineServerView;
+  private SegmentMetadataQueryConfig segmentMetadataQueryConfig;
+  private DateTime testTime;
 
   @Inject
   public ClientInfoResource(
       InventoryView serverInventoryView,
-      TimelineServerView timelineServerView
+      TimelineServerView timelineServerView,
+      SegmentMetadataQueryConfig segmentMetadataQueryConfig,
+      DateTime testTime
```

Comment:
Ok, it can start up, but it will use DateTime.now forever and ever once it does start up. It would be better to make getCurrentTime() protected and override it in the tests.